update to allow 4 character specifiers in filenames for post weights

DeniseWorthen opened the pull request:
## DESCRIPTION OF CHANGES: 

Change 3-character descriptions of destination grids for post weight generation to be 4-character by adding a trailing 0 where needed

## TESTS CONDUCTED: 
If there are changes to the build or source code, the tests below must be conducted. Contact a repository manager if you need assistance.

- [x] Compile branch on all Tier 1 machines using Intel (Orion, Jet, Hera and WCOSS2).
- [x] Compile branch on Hera using GNU. Done using 7839434.
- [x] Compile branch in 'Debug' mode on WCOSS2. Done on Dogwood using 7839434.
- [x] Run unit tests locally on any Tier 1 machine. Done on Hera using 7839434
- [x] Run relevant consistency tests locally on all Tier 1 machine.

Baselines will need to be regenerated because of the change in file names compared, e.g. ``rect.1p0_SCRIP.nc`` and ``tripole.mx025.Ct.to.rect.0p5.bilinear.nc`` will now have names containing ``1p00`` and ``0p50`` , respectively and the regression tests will flag these as missing files. Spot checking of the files confirms that they are in fact identical.


- fixes #882

DeniseWorthen committed:
update to allow 4 character specifiers in filenames

* change 3-character descriptions of destination grids for
post weight generation to be 4-character by adding a trailing
0 where needed

DeniseWorthen committed:
add orion logs

baseline location /work/noaa/stmp/dworthen/CPLD_GRIDGEN/BASELINE

DeniseWorthen commented:
@GeorgeGayno-NOAA I'm having trouble on jet. When I use ``./link_fixdirs.sh emc jet`` I get the following

```
lrwxrwxrwx 1 Denise.Worthen h-nems   53 Dec 15 20:02 am -> /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/am/20220805
-rwxr-xr-x 1 Denise.Worthen h-nems 1964 Dec 15 15:57 link_fixdirs.sh
lrwxrwxrwx 1 Denise.Worthen h-nems   55 Dec 15 20:02 orog -> /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/orog/20231027
lrwxrwxrwx 1 Denise.Worthen h-nems   50 Dec 15 20:02 orog_raw -> /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/raw/orog
lrwxrwxrwx 1 Denise.Worthen h-nems   60 Dec 15 20:02 sfc_climo -> /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/sfc_climo/20230925
```

But the orog directory isn't present. I see

```
ls /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/orog
20220805  20230615
```

But the link is for 20231027

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA I'm having trouble on jet. When I use `./link_fixdirs.sh emc jet` I get the following
> 
> ```
> lrwxrwxrwx 1 Denise.Worthen h-nems   53 Dec 15 20:02 am -> /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/am/20220805
> -rwxr-xr-x 1 Denise.Worthen h-nems 1964 Dec 15 15:57 link_fixdirs.sh
> lrwxrwxrwx 1 Denise.Worthen h-nems   55 Dec 15 20:02 orog -> /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/orog/20231027
> lrwxrwxrwx 1 Denise.Worthen h-nems   50 Dec 15 20:02 orog_raw -> /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/raw/orog
> lrwxrwxrwx 1 Denise.Worthen h-nems   60 Dec 15 20:02 sfc_climo -> /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/sfc_climo/20230925
> ```
> 
> But the orog directory isn't present. I see
> 
> ```
> ls /lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix/orog
> 20220805  20230615
> ```
> 
> But the link is for 20231027

We are implementing new 'orog' directories on all machines. Jet has not been updated yet. Can you use an older version of 'orog' for your tests?

DeniseWorthen commented:
@GeorgeGayno-NOAA I think I should be able to use the older directories. I just need the CX.mosaic.nc files I think. I'll give it a try.

DeniseWorthen committed:
add hera.intel log

*baseline located at /scratch1/NCEPDEV/stmp4/Denise.Worthen/CPLD_GRIDGEN/BASELINE

DeniseWorthen committed:
add wcoss2 log

* copy of baseline saved in
  /lfs/h2/emc/nems/noscrub/denise.worthen/CPLD_GRIDGEN

DeniseWorthen committed:
add jet log

* baselines located in /lfs4/HFIP/h-nems/Denise.Worthen/CPLD_GRIDGEN/BASELINE
* ran using fix/orog 20230615

DeniseWorthen commented:
@GeorgeGayno-NOAA I've run the RTs and created new baselines on all platforms:

```
orion: /work/noaa/stmp/dworthen/CPLD_GRIDGEN/BASELINE
hera: /scratch1/NCEPDEV/stmp4/Denise.Worthen/CPLD_GRIDGEN/BASELINE
wcoss2: /lfs/h2/emc/nems/noscrub/denise.worthen/CPLD_GRIDGEN
jet: /lfs4/HFIP/h-nems/Denise.Worthen/CPLD_GRIDGEN/BASELINE
```

GeorgeGayno-NOAA commented:
Looks good. Will merge.
